### PR TITLE
[CODEGEN] add fp16 and fp64 enable pragma for opencl

### PR DESCRIPTION
**Pull request by merrymercy**
#680

### CODE REVIEW

File: src/codegen/codegen_opencl.cc [hunk 1]
```diff
@@ -30,6 +31,35 @@ void CodeGenOpenCL::AddFunction(LoweredFunc f) {
   CodeGenC::AddFunction(f);
 }
 
+std::string CodeGenOpenCL::Finish() {
+  // inject enable pragma for fp16 and fp64
+  if (enable_fp16) {
```
Comment (tqchen) [inline src/codegen/codegen_opencl.cc]: use underscore suffix for private member

**Comment by merrymercy** [comment]
fixed

File: src/codegen/codegen_opencl.cc [hunk 2]
```diff
@@ -30,6 +31,35 @@ void CodeGenOpenCL::AddFunction(LoweredFunc f) {
...
+           "#pragma OPENCL EXTENSION cl_khr_fp16 : enable\n"
+           "#elif defined(cl_amd_fp16)\n"
+           "#pragma OPENCL EXTENSION cl_amd_fp16 : enable\n"
+           "#else\n"
+           "#error \"Half precision floating point not supported"
+                    "by OpenCL implementation on your device.\" \n"
+           "#endif\n\n";
+  }
+
+  if (_enable_fp64) {
```
Comment (tqchen) [inline src/codegen/codegen_opencl.cc]: The google c style convention puts underscore in the end

File: src/codegen/codegen_opencl.cc [hunk 3]
```diff
@@ -14,6 +14,7 @@ namespace codegen {
 
 CodeGenOpenCL::CodeGenOpenCL() {
   restrict_keyword_ = "restrict";
+  _enable_fp16 = _enable_fp64 = false;
```
Comment (tqchen) [inline src/codegen/codegen_opencl.cc]: Cxx11 encourages put simple initializer in the variable itself

**Pull request merged**